### PR TITLE
Added examples for ceu and interrupts in Arduino

### PR DESCRIPTION
**Pull request by guilhermesimas**
Added an example of ceu-arduino which ilustrates internal and external events behaviour.
It consists of a RGB LED that constantly blinks in one of three colors (Red/Green/Blue). 
Pressing the button will alternate between colors.

Added an example of interrupts in arduino.
It consists of a game involving a sensor (a trimpot was used when testing), a button and a serial interface.
The user must place the sensor trying to guess the correct position, pressing the button to submit the guess.
The game will give feedback accordingly. User must win within the specified time and attempts limit.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
-CEU_DIR  = $(error set absolute path to "<ceu>" repository)
+CEU_DIR  = $(/home/gsimas/Documents/GSoC/ceu-arduino)
```
Comment (fsantanna) [inline Makefile]: don't change this file

File: include/arduino/arduino.ceu [hunk 2]
```diff
@@ -11,6 +11,7 @@ native/const
     _INPUT,
     _INPUT_PULLUP,
     _OUTPUT,
+    _DEFAULT
```
Comment (fsantanna) [inline include/arduino/arduino.ceu]: sort by name

File: include/arduino/avr.ceu [hunk 3]
```diff
@@ -10,6 +10,7 @@ native/const
     _TIMER1_OVF_vect_num,
     _USART_RX_vect_num,
     _USART_TX_vect_num,
+    _ADC_vect_num
```
Comment (fsantanna) [inline include/arduino/avr.ceu]: sort by name

File: include/arduino/isr/analog.ceu [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+native/pre do
+	##define CEU_FEATURES_ISR_SLEEP
+end
+
+native _an_req,
+		_analogRead_read;
+native/pre do
+
+	// bit operations, set_bit and clear_bit
+	##ifndef cbi
```
Comment (fsantanna) [inline include/arduino/isr/analog.ceu]: Arduino has `bitSet`/`bitClear`, would that work?

File: include/arduino/isr/analog.ceu [hunk 5]
```diff
@@ -0,0 +1,67 @@
+#include "arduino/avr.ceu"
+#include "arduino/arduino.ceu"
+
+// Event that carries analog value
+input int AN;
+
+native/pre do
+	##define CEU_FEATURES_ISR_SLEEP
```
Comment (fsantanna) [inline include/arduino/isr/analog.ceu]: use 4 spaces for indentation

File: samples/analog-01.ceu [hunk 6]
```diff
@@ -0,0 +1,25 @@
+native/pre do
+    ##define CEU_FEATURES_ISR_SLEEP
+end
+
+#include "arduino/arduino.ceu"
+#include "arduino/isr/analog.ceu"
+#include "arduino/isr/timer.ceu"
+
+output int PIN_13;
```
Comment (fsantanna) [inline samples/analog-01.ceu]: Let's talk about these names... Maybe we'll have to change `PIN_XX` to `DIG_XX` and end up with `DIG_XX`, `PWM_XX`, and `ANA_XX`.

File: samples/isr-01.ceu [hunk 7]
```diff
@@ -14,10 +14,24 @@ end
 input  int PIN_02;
 output int PIN_13;
 
+input  int AN_01;
+output void AN_REQ_01;
+
+native _an_req;
```
Comment (fsantanna) [inline samples/isr-01.ceu]: Let's talk about changing `_an_req` to `emit ANA_XX_REQ`.

File: samples/position_game.ino [hunk 8]
```diff
@@ -0,0 +1,154 @@
+/*
```
Comment (fsantanna) [inline samples/position_game.ino]: This should not go to the main repo.

File: samples/analog-01.ceu (hunk 6 above)
Comment (guilhermesimas) [inline samples/analog-01.ceu]: Yes, I agree. Could also follow Arduino's covention and use `PIN_XX` for digital and `PIN_AX` for analog.

File: samples/isr-01.ceu (hunk 7 above)
Comment (guilhermesimas) [inline samples/isr-01.ceu]: I agree. We'll discuss the approach to deal with specific analog Pins requests and events.

**Pull request merged**